tools/generate_valve_textures.py

```
@@ -0,0 +1,249 @@
+def mix(a, b, t):
+    return tuple(round(a[k] * (1 - t) + b[k] * t) for k in range(3))
+
+
+def tint(element, color):
+    """Multiply-tint the grayscale element by `color`, preserving its shading."""
+    cr, cg, cb = color
+    out = bytearray(element)
+    for i in range(0, len(element), 4):
+        if element[i + 3] == 0:
+            continue
+        l = element[i] * 0.2126 + element[i + 1] * 0.7152 + element[i + 2] * 0.0722
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)